server.js

```diff
@@ -25,10 +45,153 @@ app.use(cors());
+
+      }});
+});
+
+// list all titles
+app.get("/titles", async (req, res) => {
+  const movies = req.query.movies
+  const tvshows = req.query.tvshows
+  let response = {
+    success: true,
+    body: {}
+  }
+  if (movies) {
+    response.body = await Title.find({ type: 'Movie' });
+    res.status(200).json(response)
+  }
+  if (tvshows) {
+    response.body = await Title.find({ type: 'TV Show' });
+    res.status(200).json(response)
+  }
+  
+  const allTheTitles = await Title.find({});
+  res.status(200).json({
+    success: true,
+    body: allTheTitles
+  });
+});
+
+// list all movies
+app.get("/titles/movies", async (req, res) => {
+  const allTheMovies = await Title.find({ type: 'Movie' });
+  res.status(200).json({
+    success: true,
+    body: allTheMovies
+  });
+});
+
+// list a specific movie
+app.get("/titles/movies/:id", async (req, res) => {
+  try {
+const singleMovie= await Title.findById(req.params.id);
+const allTheMovies = await Title.find({ type: 'Movie' });
+  if (singleMovie && allTheMovies) {
+    res.status(200).json({
+      success: true,
+      body: singleMovie
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      body: {
+        message: "Item not found"
+      }
+    });
+  }
+} catch(error) {
+  res.status(400).json({
+    success: false,
+    body: {
+      message: "Invalid id"
+    }
+  });
+}
+})
+
+// list all tv-shows
+app.get("/titles/tvshows", async (req, res) => {
+  const allTheTvShows = await Title.find({ type: 'TV Show' });
+  res.status(200).json({
+    success: true,
+    body: allTheTvShows
+  });
 });
 
+// list a specific tv-show
+app.get("/titles/tvshows/:id", async (req, res) => {
+    try {
+  const singleTvShow = await Title.findById(req.params.id);
+  const allTheTvShows = await Title.find({ type: 'TV Show' });
+    if (singleTvShow && allTheTvShows) {
+      res.status(200).json({
+        success: true,
+        body: singleTvShow
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Item not found"
+        }
+      });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+        message: "Invalid id"
+      }
+    });
+  }
+})
+
+
+// see the x latest titles (only title name, release year and type)
+//e.g. http://localhost:8080/titles/latest?limit=10
+app.get("/titles/latest", async (req, res) => {
+  const {limit} = req.query
+  const latestTitles = await Title.find({}).limit(limit).sort({release_year: -1}).select({title: 1, release_year: 1, type: 1})
+  res.status(200).json({
+    success: true,
+    body: latestTitles
+  });
+});
+
+
+
+app.get("/titles/latestMovies", async (req, res) => {
+
+  const latestMovieTitles = await Title.aggregate( [
+  {
+     $match: { type: "Movie" }
+  },
+  {
+     $group: { _id: "$name", duration: { $sum: "$duration" } }
+  }
+] )
+  res.status(200).json({
+    success: true,
+    body: latestMovieTitles
+  });
+});
```

Comment:
Did you get this to work or is it something you experimented on?
It seems interesting so tried to get it to work but couldn't and found no information about it in you firstpage with the endpoints.